### PR TITLE
Add malaysian flavor

### PR DESCRIPTION
**Pull request by SyafiqTermizi**
**Please replace these instructions with a description of your change. The
'New Fields Only' section should be removed if your pull request
doesn't add any new fields.**

Thanks for your contribution!

A checklist is included below which helps us keep the code contributions
consistent and helps speed up the review process. You can add additional
commits to your pull request if you haven't met all of these points on your
first version.

**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

**New Fields Only**

- [x] Prefix the country code to all fields.

- [x] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [ ] Prefer '<country code>PostalCodeField' for postal codes as it's
      international English; ZipCode is a term specific to the United
      States postal system.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [x] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [x] Add documentation for all fields.

### CODE REVIEW

File: docs/localflavor/my.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Malaysia (``my``)
+===============
```
[the remark this answers is not in the document]
Comment (bclermont) [reply docs/localflavor/my.rst]: @SyafiqTermizi please fix this, once merged I will add a Malaysian state field

**Comment by SyafiqTermizi** [comment]
Hi, could someone tell me why is it failing on python 3.5

**Comment by claudep** [comment]
Don't worry, that's the prospector job, that's probably a totally unrelated issue.

**Comment by bclermont** [comment]
> ImportError: No module named 'pylint.exceptions'

look like a version mismatch with prospector dependency

**Comment by benkonrath** [comment]
I just fixed the import error failure.

@SyafiqTermizi Could you rebase this PR when you have a chance? Thanks!

**Comment by SyafiqTermizi** [comment]
@benkonrath I have rebased my PR 😄

**Comment by benkonrath** [comment]
@SyafiqTermizi Thanks and sorry for the delay. I'll be able to do a full review later in the week - hopefully on Wednesday or Thursday.

**Comment by SyafiqTermizi** [comment]
Hi, @benkonrath , I do agree with your points. So I've updated my PR based on your suggestion. Hopefully, this PR can be merged soon 😄

File: localflavor/my/forms.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def to_python(self, value):
+        value = super().to_python(value)
+
+        if value in self.empty_values:
+            return self.empty_value
+        return nric.compact(value)
+
+    def prepare_value(self, value):
+        value = super().prepare_value(value)
+        return nric.format(value)
```
Comment (benkonrath) [inline localflavor/my/forms.py]: You should deal with the case of value being `None`. See an example in the IBAN form:
https://github.com/django/django-localflavor/blob/master/localflavor/generic/forms.py#L96

**Comment by benkonrath** [comment]
Looks good. Thanks!
